src/workflow_tasks.rs

```diff
@@ -554,11 +604,26 @@ impl WorkflowTaskManager {
         }
     }
 
+    /// Must be called after *every* activation is replied to, regardless of whether or not we
+    /// had some issue reporting it to server or anything else. This upholds the invariant that
+    /// every activation we issue to lang has exactly one reply.
+    ///
+    /// Any subsequent action that needs to be taken will be created as a new activation
+    pub(crate) fn activation_done(&self, run_id: &str) {
+        if self.outstanding_activations.remove(run_id).is_some() {
+            let _ = self
+                .workflow_activations_update
+                .send(WfActivationUpdate::NewPendingActivation);
+        }
+    }
+
     fn insert_outstanding_activation(&self, act: &WfActivation) {
```

Comment:
Will there ever be more than one activation per run_id?
Should this method assert there isn't?

Reply:
Good call. Started doing this and it revealed a bug where we would overwrite an eviction with new work from the server.